cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java

```diff
@@ -231,6 +231,9 @@ public void update(HttpRequest request, HttpResponder responder, @PathParam("nam
   private DatasetInstanceConfiguration  getInstanceConfiguration(HttpRequest request) {
     Reader reader = new InputStreamReader(new ChannelBufferInputStream(request.getContent()), Charsets.UTF_8);
     DatasetInstanceConfiguration creationProperties = GSON.fromJson(reader, DatasetInstanceConfiguration.class);
+    if (creationProperties.getProperties() == null) {
```

Comment:
You can simply have `DatasetInstanceConfiguration.getProperties()` implemented as:

```java
public Map<String, String> getProperties() {
  return properties == null ?? ImmutableMap.<String, String>of() : properties;
}
```